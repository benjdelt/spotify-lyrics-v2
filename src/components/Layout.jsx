function Layout(props) {
  return (
    <>
      <header className="app-header">
      <div className="logo">
        <h1>
          <img src="inverted-logo.png" alt="logo"/>
          <span className="menu-text">Spotify Lyrics</span>
        </h1>
      </div>
      {props.nav}
      </header>
      <main className="app-main">
        {props.children}
      </main>
    </>
  )
}

export default Layout;
