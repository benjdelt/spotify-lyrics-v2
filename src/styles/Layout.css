.app-header {
  background-color: #121212;
  height: 100vh;
  width: 22em;
  position: fixed;
  display: flex;
  flex-direction: column;
}

.app-header>* {
  padding: 0 1.5rem;
}

.logo img {
  width: 1.5em;
  margin-right: .5em;
}

.logo {
  text-decoration: none;
}

h1 {
  color: #fff;
  font-size: 2rem;
  display: flex;
  align-items: center;
}

@media (max-width: 1024px) {
  .app-header {
    width: 21em;
  }
}

@media (max-width: 860px) {
  .app-header {
    height: 5em;
    width: 100vw;
    flex-direction: row;
    bottom: 0;
  }

  .logo {
    display: flex;
  }

  h1 .menu-text {
    display: none;
  }
}

@media (max-width: 415px) {
  .app-header>* {
    padding-left: .5em;
  }
}

@media (max-width: 325px) {
  .app-header>* {
    padding: 0;
  }

  .logo {
    padding-left: .5em;
  }
}
