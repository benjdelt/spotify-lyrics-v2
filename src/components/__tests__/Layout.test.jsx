import { render, cleanup } from '@testing-library/react';
import Layout from '../Layout';

afterEach(cleanup)

it('should take a snapshot', () => {
  const { asFragment } = render(<Layout />)
  expect(asFragment(<Layout />)).toMatchSnapshot()
});
