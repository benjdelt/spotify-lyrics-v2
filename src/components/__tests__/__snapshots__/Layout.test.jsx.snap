// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <header
    class="app-header"
  >
    <div
      class="logo"
    >
      <h1>
        <img
          alt="logo"
          src="inverted-logo.png"
        />
        <span
          class="menu-text"
        >
          Spotify Lyrics
        </span>
      </h1>
    </div>
  </header>
  <main
    class="app-main"
  />
</DocumentFragment>
`;
